src/org/dux/stracetool/Tracer.java

```diff
@@ -0,0 +1,78 @@
+package org.dux.stracetool;
+
+import org.dux.cli.DuxCLI;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Tracer {
+    // TODO Allow for arbitrary temp file names (need to sanitize in constructor)
+    private static final String TMP_FILE = ".dux_out";
+    private String[] args;
+
+    private static final String[] STRACE_CALL = {
+            "strace",
+            "-o", TMP_FILE               // write to tmp file
+    };
+
+    private static final String[] PROCMON_CALL = {
+            "TODO implement"
+    };
+
+    public Tracer(List<String> args) {
+        String os = System.getProperty("os.name");
+        List<String> argList = null;
+        if (os.startsWith("Linux")) {
+            argList = new ArrayList<>(Arrays.asList(STRACE_CALL));
+        } else if (os.startsWith("Windows")) {
+            // TODO implement
+            argList = new ArrayList<>(Arrays.asList(PROCMON_CALL));
+        } else {
+            // TODO What to do here?
+        }
+        System.out.println("args = " + args);
+        argList.addAll(args);
+        System.out.println("args = " + argList);
```

Comment:
Use the logger instead of `System.out` (I don't know if you copied this code or not, but we should try to increase consistency so that when I fix the logging it's fixed everywhere).